WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift

```diff
@@ -58,8 +58,11 @@ struct PointOfSaleDashboardView: View {
         .animation(.easeInOut(duration: Constants.connectivityAnimationDuration), value: viewModel.showsConnectivityError)
         .background(Color.posPrimaryBackground)
         .navigationBarBackButtonHidden(true)
-        .sheet(item: $totalsViewModel.cardPresentPaymentOnboardingViewModel) { viewModel in
+        .posModal(item: $totalsViewModel.cardPresentPaymentOnboardingViewModel, onDismiss: {
+            totalsViewModel.cancelOnboarding()
+        }) { viewModel in
             paymentsOnboardingView(from: viewModel)
+                .posInteractiveDismissDisabled(true)
```

Comment:
I'm confused by this... in the description it says we should be able to tap outside the modal to dismiss it, but this modifier prevents that happening.

Payment modals can be dismissed by tapping outside, which was a design request, so I think onboarding ones should work the same way.

You could extract the frame code from `PointOfSaleCardPresentPaymentAlert` to share it between payment alerts and onboarding – they should probably have the same size frames for visual consistency. That might be something to do once we redesign them though, the old style may not fit because it's so wordy.

```suggestion
```

Reply:
> in the description it says we should be able to tap outside the modal to dismiss it, but this modifier prevents that happening.

Thanks for catching this one, I originally meant "tapping outside the modal should **not** dismiss it" so it's very confusing 😓 

> Payment modals can be dismissed by tapping outside, which was a design request, so I think onboarding ones should work the same way.

I see, yea it makes sense to make it possible to dismiss the onboarding by tapping outside if there was a design request. For more context, the interactive dismissal was first disabled with the feedback in https://github.com/woocommerce/woocommerce-ios/pull/14185#discussion_r1812064752 to prevent the onboarding from being dismissed accidentally. But from more testing, the webview that shows the WooPayments/Stripe form can still be dismissed interactively and some changes are required within the onboarding view which affects the pre-existing onboarding flows.

Reply:
Interactive dismissal enabled in https://github.com/woocommerce/woocommerce-ios/pull/14224/commits/9da408ad5227ae23f8c664de6f7a64314948c80f.

> You could extract the frame code from PointOfSaleCardPresentPaymentAlert to share it between payment alerts and onboarding – they should probably have the same size frames for visual consistency. That might be something to do once we redesign them though, the old style may not fit because it's so wordy.

Good to know, added in https://github.com/woocommerce/woocommerce-ios/pull/14224/commits/107699e9bae23118124117a6f22bfe99de891d0b for POS onboarding and the replacement for POS payment modals is in https://github.com/woocommerce/woocommerce-ios/pull/14224/commits/1b7dcfc847421fa54d6aceb542b6db3eab0501a9.